File: sast-js/case/accuracy/object_sensitive/collection/map_object_sensitive_002_F.js
// evaluation information start
// real case = false
// evaluation item = 准确度->对象敏感与域敏感分析->区分字典/列表/数组的不同元素
// scene introduction = 字典->字典对象
// level = 2
// bind_url = accuracy/object_sensitive/collection/map_object_sensitive_002_F
// evaluation information end
const { execSync } = require('child_process');


function map_object_sensitive_002_F(__taint_src) {
  let map = new Map();
  map.set("key1", __taint_src);
  let m = new Map();
  m.set("key1", "_");
  __taint_sink(m);
}

function __taint_sink(o) {
  execSync(o);
}

const taint_src = "taint_src_value";

map_object_sensitive_002_F(taint_src);

File: sast-python2/case/accuracy/object_sensitive/collection/map_object_sensitive_002_F.py
# -*- coding: utf-8 -*-
# evaluation information start
# real case = false
# evaluation item = 准确度->对象敏感与域敏感分析->区分字典/列表/数组的不同元素
# scene introduction = 字典->字典对象->value为source
# level = 2
# bind_url = accuracy/object_sensitive/collection/map_object_sensitive_002_F
# evaluation information end
import os

def map_object_sensitive_002_F(taint_src):
    map = {}
    map[u"key1"] = taint_src

    m = {}
    m[u"key1"] = u"_"

    taint_sink(m)


def taint_sink(o):
    os.system(unicode(o))


# 示例调用
if __name__ == u"__main__":
    taint_src = u"taint_src_value"
    map_object_sensitive_002_F(taint_src)

